### PR TITLE
implemented button to copy definition link

### LINKED ISSUE #417: FEATURE: make definitions shareable
### Description

Currently, users can only lookup definitions, but it would also be great if there's a shareable link that users can share with others so they could easily view the definition instead of manually searching again

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by trp02**
Closes #417 

### What new changes did you make? Tick all applicable boxes
- [ ] Added new abbreviation
- [ ] Fixed something in the source code
- [x] Added a new feature
- [ ] Fixed the docs (README.md, CONTRIBTUING.md, etc)

Created the button to copy the link and share definition. Tried to stay consistent with the rest of the website in terms of design. Let me know if you want to change. 
### Share a screenshot of new changes
image

### CODE REVIEW

File: src/components/form.jsx [hunk 1]
```diff
@@ -143,22 +144,40 @@ const Form = () => {
             </button>
           </form>
 
+          <style>
```
Comment (Njong392) [inline src/components/form.jsx:147]: oof uhm... this project uses tailwind, and I'm sure there is some way to apply these classes to the required button using the tailwind way. Why go this route? I like the output, but this in the middle of the code just makes it look out of place.

**Comment by trp02** [comment]
Apologies. I hadn't worked with tailwind before so it seemed a little messier at first but certainly is a lot more concise! It seems to be working as expected on the preview. LMK if anything else needs changing.

**Comment by Njong392** [comment]
This looks splendid! Thank you so much for this feature.One tiny thing, and I hope I'm not nagging hehe. When someone copies, the link, for me it doesn't show like a 'copied to clipboard' or anything to indicate that I did copy the link for that abbreviation. Is there something we could do to fix this?

**Comment by Njong392** [comment]
I still feel it's not obvious enough that the link icon is for copying an abbreviation. Instead of changing to another icon, what if we could make it simple by adding some indicative text that says 'copied'? So when the user clicks the link, just like you've implemented for the icon to change, let that small text pop up instead and then you use `setTimeOut` like you did to replace the text after a few secs. I'm also open to any better suggestions if you have

**Comment by trp02** [comment]
Is this closer to what you were thinking?

**Comment by Njong392** [comment]
Yes, this is perfect! I no longer want to bother you with this PR, seeing as you've done so much work on it. I'll merge this, and then we can always fix alignment issues etc as some other contribution. Thank you very much again for the neat feature!

**Comment by trp02** [comment]
Sounds good, I will revisit the code over the weekend with a fresh mind and cleanup anything I can find. I learned a lot so thank you for the opportunity!

**Pull request merged**